WcaOnRails/spec/controllers/competitions_controller_spec.rb

```diff
@@ -495,6 +495,16 @@
         end.to change { enqueued_jobs.size }.by(1)
       end
 
+      it "notifies organizers correctly when id changes" do
+        new_organizer = FactoryBot.create :user
+        old_id = competition.id
+        competition.id = "NewId2018"
+        expect(CompetitionsMailer).to receive(:notify_organizer_of_addition_to_competition).with(competition.delegates.last, competition, new_organizer).and_call_original
+        expect do
+          patch :update, params: { id: old_id, competition: { id: "NewId2018", organizer_ids: new_organizer.id } }
+        end.to change { enqueued_jobs.size }.by(1)
+      end
```

Comment:
This was quite tricky to write as we specify what `competition` we would like the mailer do receive, but this competition gets changed by the request directly (`id` in our case). That's why I set the `id` explicitly before passing it to the mailer assertion.